### PR TITLE
Redesign UI

### LINKED ISSUE #29: UI looks terrible on mobile
The UI is clearly not made to be rendered on mobile. See screenshots.

image
image

### PR DESCRIPTION
**Pull request by alexmshepherd**
This is a complete UI overhaul. @4censord what do you think? Closes #29.

### CODE REVIEW

**Comment by alexmshepherd** [comment]
This is live at [the canonical instance](https://owdex.com/) right now.

**Comment by 4censord** [comment]
I will get to reviewing this in detail at the earliest on Tuesday.

On first look I like it.

**Review by 4censord — CHANGES_REQUESTED** [review]
We should talk about tabs/spaces in the HTML templates. I don't care about which one we do, is just should be consistent across files.

File: owdex/linkmanager.py [hunk 1]
```diff
@@ -83,12 +84,20 @@ def search(self, query, indices):
...
+                for result in index_results:
+                    # we add the index attribute so we can show the index this result was pulled from
+                    result.update({"index": index_name})
+                results.extend(index_results)
+
+        except SolrError as e:
+            if "org.apache.solr.search.SyntaxError" in str(e):
+                pass  # it's a malformed query, just return what we have -- typically an empty array
+            else:
+                raise
```
[the remark this answers is not in the document]
Comment (alexmshepherd) [reply owdex/linkmanager.py:100]: Yep, agreed.

File: owdex/users.py [hunk 2]
```diff
@@ -10,36 +10,32 @@
...
 
-
-@users_bp.route("/signup", methods=["GET", "POST"])
-@app.limiter.limit(
-    "5/minute;10/hour;25/day",
-    scope="users",
-)
-def signup():
-    if f.request.method == "POST":
+    elif action == "signup":
```
[the remark this answers is not in the document]
Comment (alexmshepherd) [reply owdex/users.py:38]: Got it. I'll try to rework this.

**Comment by alexmshepherd** [comment]
> Also, imo there should be a backlink from the error pages to either the homepage or the page from which they originally came

Agreed.

**Comment by alexmshepherd** [comment]
> tabs/spaces in the HTML templates

I wasn't aware this wasn't consistent. I'll work on improving pre-commit. My preference is for tabs.

File: owdex/users.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexmshepherd) [reply owdex/users.py]: Resolved by [f442bdd](https://github.com/owdex/owdex/pull/57/commits/f442bdd695446243c8dcb8503c4d4ce9b92bb8d2).

File: owdex/linkmanager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexmshepherd) [reply owdex/linkmanager.py]: Resolved by [0cec78f](https://github.com/owdex/owdex/pull/57/commits/0cec78f1e141e7412abda4bc4497f182c457ca2a).

File: owdex/users.py
Comment (4censord) [inline owdex/users.py:1]: I like this more, it feels cleaner. 
Thank you

**Review by 4censord — APPROVED** [review]

**Comment by alexmshepherd** [comment]
@4censord See #69 (yes I know).

**Pull request merged**